pkg/sql/create_table.go

```diff
@@ -176,13 +176,16 @@ func (n *createTableNode) startExec(params runParams) error {
 		telemetry.Inc(sqltelemetry.CreateTempTableCounter)
 	}
 
-	// Guard against creating non-partitioned indexes on a partitioned table,
+	// Warn against creating non-partitioned indexes on a partitioned table,
 	// which is undesirable in most cases.
-	if params.SessionData().SafeUpdates && n.n.PartitionBy != nil {
+	if n.n.PartitionBy != nil {
 		for _, def := range n.n.Defs {
 			if d, ok := def.(*tree.IndexTableDef); ok {
 				if d.PartitionBy == nil {
-					return pgerror.DangerousStatementf("non-partitioned index on partitioned table")
+					params.p.SendClientNotice(
+						params.ctx,
+						pgerror.Noticef("creating non-partitioned index on partitioned table may not be performant"),
```

Comment:
please provide a hint too (`errors.WithHint`) as to what should be done instead.

(Preferably with a doc link - although I think we need a separate issue+PR to introduce a standard way to populate doc links in error messages. We don't have that infrastructure right now. The nearest we have is the doc links in the sql.y.)